src/App.test.js

```diff
@@ -653,12 +666,104 @@ describe('App component', () => {
         expect(dopplerOriginDate).toBeDefined();
         expect(dopplerOriginDate.getFullYear()).toBeGreaterThan(2018);
       });
     });
 
+    it('should check utm parameters are stored', async () => {
+      // Arrange
+      const dependencies = {
+        sessionManager: createDoubleSessionManager(),
+        localStorage: createLocalStorageDouble(),
+        dopplerSitesClient: dopplerSitesClientDouble,
+      };
+
+      createJsonParse();
+
+      // Act
+      act(() => {
+        render(
+          <AppServicesProvider forcedServices={dependencies}>
+            <Router
+              initialEntries={[
+                '/signup?origin=testOrigin&utm_source=test&utm_campaign=testcampaign&utm_medium=testmedium&utm_term=testterm',
+              ]}
+            >
+              <App locale="en" />
+            </Router>
+          </AppServicesProvider>,
+        );
+      });
+
+      // Assert
+      const localStorageItems = dependencies.localStorage.getAllItems();
+      await waitFor(() => {
+        expect(localStorageItems['UtmCookies']).toBeDefined();
+        expect(localStorageItems['UtmCookies']).toMatch('UTMSource');
+        expect(localStorageItems['UtmCookies']).toMatch('test');
+        expect(localStorageItems['UtmCookies']).toMatch('UTMCampaign');
+        expect(localStorageItems['UtmCookies']).toMatch('testcampaign');
+        expect(localStorageItems['UtmCookies']).toMatch('UTMMedium');
+        expect(localStorageItems['UtmCookies']).toMatch('testmedium');
+        expect(localStorageItems['UtmCookies']).toMatch('UTMTerm');
+        expect(localStorageItems['UtmCookies']).toMatch('testterm');
+      });
+    });
+
+    it('should accumulate utm parameters when there was a previous navigation', async () => {
+      // Arrange
+      const dependencies = {
+        sessionManager: createDoubleSessionManager(),
+        localStorage: createLocalStorageDouble(),
+        dopplerSitesClient: dopplerSitesClientDouble,
+      };
+
+      const utmCookie = {
+        date: new Date().toISOString(),
+        UTMSource: 'utmsource1',
+        UTMCampaign: 'utmcampaign1',
+        UTMMedium: 'utmmedium1',
+        UTMTerm: 'utmterm1',
+      };
+      dependencies.localStorage.setItem('UtmCookies', JSON.stringify(utmCookie));
+      createJsonParse(utmCookie);
+      // Act
+      act(() => {
+        render(
+          <AppServicesProvider forcedServices={dependencies}>
+            <Router
+              initialEntries={[
+                '/signup?origin=testOrigin&utm_source=test&utm_campaign=testcampaign&utm_medium=testmedium&utm_term=testterm',
+              ]}
+            >
+              <App locale="en" />
+            </Router>
+          </AppServicesProvider>,
+        );
+      });
+
+      // Assert
+      const localStorageItems = dependencies.localStorage.getAllItems();
+      await waitFor(() => {
+        expect(localStorageItems['UtmCookies']).toBeDefined();
```

Comment:
You could do this here:
```suggestion
        expect(localStorageItems['UtmCookies']).toBeDefined();
        const utmCookiesObject = parse(localStorageItems['UtmCookies']);
        expect(utmCookiesObject.length).toBeGreaterThan(1);
```

Where parse would be a copy of JSON.parse, before doing the mock:
```
const parse = JSON.parse;
```

Reply:
changed!